viewer/packages/data-providers/src/image-360-data-providers/descriptor-providers/datamodels/cdm/Cdf360CdmDescriptorProvider.ts

```diff
@@ -0,0 +1,251 @@
+    });
+
+    const groups = groupBy(imagesWithStation, imageResult => {
+      const station = imageResult.image.properties.cdf_cdm['Cognite360Image/v1'].station360 as InstanceIdentifier;
+      return `${station.externalId}-${station.space}`;
+    });
+
+    const ret = Object.values(groups)
+      .concat(imagesWithoutStation.map(p => [p]))
+      .map(imageWithFileDescriptors => {
+        return this.getHistorical360ImageSet(collectionId, collectionLabel, imageWithFileDescriptors);
+      });
+
+    return ret;
+  }
+
+  private async queryCollection({
+    image360CollectionExternalId,
+    space
+  }: Image360DataModelIdentifier): Promise<ExhaustedQueryResult> {
+    const result: ExhaustedQueryResult = {
+      image_collection: [],
+      images: [],
+      stations: []
+    };
+
+    const query = get360CollectionQuery(image360CollectionExternalId, space);
+
+    const imageLimit = query.with.images.limit;
+
+    let nextCursor: QueryNextCursors<Cdf360FdmQuery> | undefined = undefined;
+    let hasNext = true;
+
+    while (hasNext) {
+      const {
+        image_collection,
+        images,
+        stations,
+        nextCursor: currentCursor
+      }: QueryResult = await this._dmsSdk.queryNodesAndEdges(query, nextCursor);
+      if (result.image_collection.length === 0) {
+        result.image_collection.push(...image_collection);
+      }
+      result.images.push(...images);
+      result.stations.push(...stations);
+
+      hasNext =
+        images.length === imageLimit && currentCursor?.images !== undefined && currentCursor?.stations !== undefined;
+      nextCursor = {
+        images: currentCursor?.images,
+        stations: currentCursor?.stations
+      };
+    }
+
+    return result;
+  }
+
+  private async getFileDescriptors(images: ImageResult) {
+    const imageProps = images.map(image => image.properties.cdf_cdm['Cognite360Image/v1']);
+    const cubeMapFileIds = imageProps.flatMap(imageProp => [
+      imageProp.front as DirectRelationReference,
+      imageProp.back as DirectRelationReference,
+      imageProp.left as DirectRelationReference,
+      imageProp.right as DirectRelationReference,
+      imageProp.top as DirectRelationReference,
+      imageProp.bottom as DirectRelationReference
+    ]);
+
+    const externalIdBatches = chunk(chunk(cubeMapFileIds, 1000), 15);
+
+    const fileInfos: FileInfo[] = [];
+    for (const parallelBatches of externalIdBatches) {
+      const batchFileInfos = (
+        await Promise.all(parallelBatches.map(batch => getCdmFiles(this._cogniteSdk, batch)))
+      ).flatMap(p => p);
+      fileInfos.push(...batchFileInfos);
+    }
+
+    return chunk(fileInfos, 6);
+  }
+
+  private getHistorical360ImageSet(
+    collectionId: string,
+    collectionLabel: string,
+    imageFileDescriptors: { image: ImageInstanceResult; fileDescriptors: FileInfo[] }[]
+  ): Historical360ImageSet {
+    const mainImageProps = imageFileDescriptors[0].image.properties.cdf_cdm['Cognite360Image/v1'];
```

Comment:
This will show same timestamp for all revision, refer [issue](https://github.com/cognitedata/reveal/pull/4886) fix

Reply:
Good catch, fixed in [b24ff79](https://github.com/cognitedata/reveal/pull/4894/commits/b24ff79808b01684a7688d8fc7e2782ba0bb5843)